### PR TITLE
feat: npm publishing only relevant files

### PR DESCRIPTION
**Pull request by danielherrerohernando**
## What’s the focus of this PR
This PR is intended to remove from the npm published package all the files that are not relevant for consumers (eslintrc, travis...) keeping optional ones like the examples folder that may be useful at some point.

If you think the systemic's npm package should contain all these development files simply reject this PR but in my humble opinion, it would better to publish a clean package containing only the files that matter.

## How to review this PR
Check how we can ~~whitelist the files we want to include in the package by using the `files `property in package.json, this way there is no need to have a npmignore file.~~ blacklist all the unnecessary files by npmignoring them.

### CODE REVIEW

**Comment by cressie176** [comment]
Hi @danielherrerohernando,

If we were to create another (necessary) file, is there anything that will remind us to add it to the white list?

**Comment by danielherrerohernando** [comment]
> Hi @danielherrerohernando,
> 
> If we were to create another (necessary) file, is there anything that will remind us to add it to the white list?

@cressie176 No, that's the downside of this approach. If you think that's going to happen quite often in the future it's better to use the `npmignore` blacklisting approach

**Comment by cressie176** [comment]
I think npmignore is a safer approach.

**Comment by danielherrerohernando** [comment]
@cressie176 ok, I've updated the PR accordingly, now just using .npmignore

File: .npmignore [hunk 1]
```diff
@@ -1,5 +1,10 @@
 node_modules
 coverage
 *.log
-.codeclimate
```
Comment (cressie176) [inline .npmignore:4]: .codeclimate needs to stay. It contains a secret key for publishing test coverage etc to code climate. Ideally this should be done by CI, but until then we need to leave it in.

**Review by cressie176 — CHANGES_REQUESTED** [review]

File: .npmignore (hunk 1 above)
Comment (danielherrerohernando) [reply .npmignore:4]: @cressie176 I see, solved!

**Pull request merged**